File: Leetcode First 200 JS solution/38. Count and Say.js
var countAndSay = function(n) {
    const f = Array.from({length: n});
    f[0] = '1';
    for(let i = 1; i < n; i ++) {
        let j = 0, str = f[i - 1], res = "";
        const sl = str.length;
        while(j < sl) {
            let c = str[j], k = 1;
            while(j + 1 < sl && str[j + 1] == str[j]) j ++, k ++;
            res += k + "" + c;
            j ++;
        }
        f[i] = res;
    }
    return f[n - 1];
};

//

File: Daily Coding Practice/2019-11-26/38 Count and Say.py
class Solution:
    def countAndSay(self, n):

        s = '1'
        for i in range(1, n):

            curr_s = ''
            j = 0
            while j < len(s):
                u = j
                # count how many same 1 or 2
                while u < len(s) and s[u] == s[j]:
                    u += 1
                curr_s += str(u-j)
                curr_s += s[j]

                # update the j
                j = u
            
            s = curr_s
        
        return s